### DISCUSSION TITLE
getFormState? Accessing latest form state from callback similar to getValues

### DISCUSSION THREAD

**Discussion by zyasir** [react-hook-form/react-hook-form#6575] [Q&A]
Is there a way to getFormState? Similar to getValues?

Here's a codesandbox demo- https://codesandbox.io/s/interesting-solomon-jbql4?file=/src/App.js:511-553
And here is video of me demoing the codesandbox - 

https://user-images.githubusercontent.com/49174024/133808216-63c8f946-dec0-4106-aba2-ab0c4ad67212.mov

You'll notice that once you focus the text input then unfocus it, it shows "Field is dirty". But then when you click submit, it logs field is not dirty.

To make this example work you have too uncomment that line above so that `isDirty` is accessed in render of where submit handler is made (app level). But this will cause it to re-render the app. I just want to access latest form state in a callback. 

Is this possible?

**Comment by adriantrunzo** [comment]
You might find this issue relevant: https://github.com/react-hook-form/react-hook-form/issues/6243.

**Reply by zyasir** [reply]
Thanks I posted asking if `getValues` then why not `getFormState`

**Comment by bluebill1049** [comment]
answered here: https://github.com/react-hook-form/react-hook-form/issues/6243#issuecomment-922122395

**Comment by aldenquimby** [comment]
@zyasir did you ever find a solution here? To avoid constant re-renders, I am currently using `getValues()` in a callback and then doing a manual deep equality comparison against `defaultValues`. But it would be nice to have some way to access `isDirty` or `dirtyFields` in a callback

**Reply by aldenquimby** [reply]
It looks like if `getDirtyFields` was exported from this lib, that would mostly solve my problem, because then I could call `getDirtyFields(defaultValues, getValues())` in the callback
